### PR TITLE
Adding LDA support and info about unused fields in local predictions

### PR DESCRIPTION
**Pull request by mmerce**
to be kept in next until LDA hits production

### CODE REVIEW

File: lib/LocalEnsemble.js [hunk 1]
```diff
@@ -198,6 +198,17 @@ if (NODEJS) {
 }
 
 
+function intersect(array1, array2) {
+    var temp;
+    if (array2.length > array1.length) {
+      temp = array2, array2 = array1, array1 = temp;
+    } // indexOf to loop over shorter
+    return array1.filter(function (e) {
+        if (array2.indexOf(e) !== -1) return true;
```
Comment (jaor) [inline lib/LocalEnsemble.js]: there's no else, that's okay? why not simply `return (array2.indexOf(e) !== -1)`?

File: lib/LocalEnsemble.js [hunk 2]
```diff
@@ -240,15 +245,51 @@ LocalEnsemble.prototype.predict = function (inputData, method, options, cb) {
...
+      mean = options.mean;
+    } else {
+      mean = false;
+    }
+    if ((typeof options !== 'undefined') &&
+        (options.hasOwnProperty('addUnusedFields'))) {
+      addUnusedFields = options.addUnusedFields;
+    } else {
+      addUnusedFields = false;
+    }
```
Comment (jaor) [inline lib/LocalEnsemble.js]: these things can be written shorter as:

```js
mean = (typeof options !== 'undefined') && options.hasOwnProperty('mean') && mean
```

but if you find your way better, it's okay with me too.

File: lib/LocalModel.js [hunk 3]
```diff
@@ -210,21 +231,36 @@ LocalModel.prototype.predict = function (inputData, missingStrategy,
...
     for (field in inputData) {
       if (inputData.hasOwnProperty(field)) {
         if (inputData[field] === null ||
             (typeof this.tree.fields[field] === 'undefined' &&
-             typeof this.invertedFields[field] === 'undefined')) {
+             typeof this.invertedFields[field] === 'undefined') ||
+             (typeof this.tree.fields[field] !== 'undefined' &&
+              this.tree.objectiveField === field) ||
+             (typeof this.invertedFields[field] !== 'undefined' &&
+              this.tree.objectiveField === this.invertedFields[field])) {
```
Comment (jaor) [inline lib/LocalModel.js]: is this complicated boolean expression the same as the one complicated expression in LocalLogisticRegression? perhaps we could put it/them in a separate little function and give it a name?

File: lib/LocalEnsemble.js (hunk 1 above)
Comment (mmerce) [inline lib/LocalEnsemble.js]: it's okay but it can be done this way too

File: lib/LocalModel.js (hunk 3 above)
Comment (mmerce) [inline lib/LocalModel.js]: there's a slight difference. The objective field and fields structure are attributes of hte local logistic regression, not of `tree`

**Pull request merged**